armi/reactor/blocks.py

```diff
@@ -1519,6 +1536,84 @@ def coords(self, rotationDegreesCCW=0.0):
+        Create a new homogenized copy of a block that is less expensive than a full deepcopy.
+
+        Notes
+        -----
+        This can be used to improve performance when a new copy of a reactor needs to be
+        built, but the full detail of the block (including component geometry, material,
+        number density, etc.) is not required for the targeted physics solver being applied
+        to the new reactor model.
+
+        The main use case is for the uniform mesh converter (UMC). Frequently, a deterministic
+        neutronics solver will require a uniform mesh reactor, which is produced by the UMC.
+        Many deterministic solvers for fast spectrum reactors will also treat the individual
+        blocks as homogenized mixtures. Since the neutronics solver does not need to know about
+        the geometric and material details of the individual child components within a block,
+        we can save significant effort while building the uniform mesh reactor with the UMC
+        by omitting this detailed data and only providing the necessary level of detail for
+        the uniform mesh reactor: number densities on each block.
+
+        .. note: Individual components within a block can have different temperatures, and this
+        can affect cross sections. This temperature variation is captured by the lattice physics
+        module. As long as temperature distribution is correctly captured during cross section
+        generation, it doesn't need to be transferred to the neutronics solver directly through
+        this copy operation.
+
+        .. note: If you make a new block, you must add it to an assembly and a reactor.
+
+        See Also
+        --------
+        armi.reactor.converters.uniformMesh.UniformMeshGeometryConverter.makeAssemWithUniformMesh
+        """
+
+        b = self.__class__(self.getName(), height=self.getHeight())
+        b.setType(self.getType())
+
+        # assign macros and LFP
+        b.macros = self.macros
+        b._lumpedFissionProducts = self._lumpedFissionProducts
+        b.p.buGroup = self.p.buGroup
+
+        hexComponent = Hexagon(
+            "homogenizedHex",
+            "Mixture",
+            self.getAverageTempInC(),
+            self.getAverageTempInC(),
+            self._pitchDefiningComponent[1],
+        )
+        hexComponent.setNumberDensities(self.getNumberDensities())
+        b.add(hexComponent)
+
+        # create a null component with cladding flags and spatialLocator from source block's
+        # clad component in case pin locations need to be known for phm ysics solver
+        # only works if there is a single clad component that has a MultiIndexLocation
+        if self.hasComponents(Flags.CLAD):
+            pinComponent = Circle(
+                "voidPin",
+                "Void",
+                self.getAverageTempInC(),
+                self.getAverageTempInC(),
+                0.0,
+            )
+            clad = self.getComponent(Flags.CLAD)
```

Comment:
This is causing an error for me.

I have blocks that have multiple components with the `CLAD` flag, because I have some assemblies that have multiple fuel types loaded into it.

Reply:
I guess you note this in a comment above...

Reply:
We spoke offline; updated to handle multiple clad components